packages/asc-core/src/utils/themeUtils.ts

```diff
@@ -20,39 +29,42 @@ export const getTypographyFromTheme = (
   return fromTheme(`typography.${[attributeType]}`)({ theme })
 }
 
-export const getFocusStyle = (theme: Theme.ThemeInterface) => css`
+export const focusStyle = () => ({
+  theme,
+}: {
+  theme: Theme.ThemeInterface
+}) => css`
   &:focus {
-    outline-color: ${getColorFromTheme(theme, 'support', 'focus')};
+    outline-color: ${color('support', 'focus')({ theme })};
     outline-style: solid;
     outline-width: medium;
   }
 `
-export const getBreakpointFromTheme = (
-  theme: Theme.ThemeInterface,
-  type: Theme.TypeBreakpoint,
-  variant: string,
-) => {
-  const breakpoint: Theme.GetBreakpointFunc = fromTheme(
+export const breakpoint = (type: Theme.TypeBreakpoint, variant: string) => ({
+  theme,
+}: {
+  theme: Theme.ThemeInterface
+}) => {
+  const breakpointFunc: Theme.GetBreakpointFunc = fromTheme(
     `breakpoints.${[variant]}`,
   )({
     theme,
   })
-  return breakpoint && breakpoint(type)
+  return breakpointFunc && breakpointFunc(type)
 }
 
-export const fillSvgFromTheme = (
-  theme: Theme.ThemeInterface,
+export const svgFill = (
   colorType?: Theme.TypeLevel,
   variant: string = 'main',
```

Comment:
we can type the variants, as we know which ones are restricted (cannot be only just any string). perhaps we can use keyof here?

Reply:
will be done in the second iteration